h/renderers.py

```diff
@@ -0,0 +1,31 @@
+# Taken from https://pyramid-cookbook.readthedocs.org/en/latest/templates/customrenderers.html
+# with minor modifications
+import csv
+try:
+    import StringIO as io # python 2
+except ImportError:
+    from io import StringIO # python 3
+
+class CSVRenderer(object):
+   def __init__(self, info):
+      pass
+
+   def __call__(self, value, system):
+      """ Returns a plain CSV-encoded string with content-type
+      ``text/csv``. The content-type may be overridden by
+      setting ``request.response.content_type``."""
+
+      request = system.get('request')
+      if request is not None:
+         response = request.response
+         ct = response.content_type
+         if ct == response.default_content_type:
+            response.content_type = 'text/csv'
+
+      fout = io.StringIO()
+      writer = csv.writer(fout, delimiter=',', quotechar=',', quoting=csv.QUOTE_MINIMAL)
+
+      writer.writerow(value.get('header', []))
+      writer.writerows(value.get('rows', []))
+
+      return fout.getvalue()
```

Comment:
I think we should probably add some tests for this class. I'd expect it to test at least:

- the default value of `content_type`
- the string output of the renderer for a simple example